error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/FileRules.java

```diff
@@ -141,4 +142,18 @@ File after(File directory, String prefix, String suffix) throws IOException {
       return Files.createTempFile(directory.toPath(), prefix, suffix).toFile();
     }
   }
+
+  /** Prefer `File#mkdirs` before `Files#exists` to avoid concurrency issues. */
+  static final class MkdirsBeforeFilesExists {
+    @BeforeTemplate
+    boolean before(Path path) {
```

Comment:
I specifically cover `Path` here, but the same could apply for `File` as well. Let me know if that should be added.